rules/numeric-separators-style.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+const {chunk, last} = require('lodash');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const getNumberLiteralType = require('./utils/get-number-literal-type');
+
+const defaultOptions = {
+	hexadecimal: {minimumThreshold: 0, preferedGroupLength: 2},
+	binary: {minimumThreshold: 0, preferedGroupLength: 4},
+	octal: {minimumThreshold: 0, preferedGroupLength: 4},
+	bigint: {minimumThreshold: 5, preferedGroupLength: 3},
+	number: {minimumThreshold: 5, preferedGroupLength: 3}
+};
+
+const literalNotations = {
+	hexadecimal: '0x',
+	binary: '0b',
+	octal: '0o',
+	bigint: 'n',
+	number: ''
+};
+
+const chunkString = (string, size) => chunk(string, size).map(elt => elt.join(''));
+
+function addLiteralNotation(notation, string) {
+	if (notation === '') {
+		return string;
+	}
+
+	return notation === 'n' ?
+		string + notation :
+		notation + string;
+}
+
+function removeLiteralNotation(notation, string) {
+	if (notation === '') {
+		return string;
+	}
+
+	return notation === 'n' ?
+		string.replace(new RegExp(notation + '$'), '') :
+		string.replace(notation, '');
+}
+
+function getChunks(string, size, restAtEnd) {
+	let chunks = chunkString(string, size);
+	const lastElement = last(chunks);
+	if (restAtEnd || chunks.length === 1 || lastElement.length === size) {
+		return chunks;
+	}
+
+	// Put the rest of the number, currently at the end, at the beginning
+	const offset = '0'.repeat(size - lastElement.length);
+	string = offset + string;
+	chunks = chunkString(string, size);
+	chunks[0] = chunks[0].replace(offset, '');
+	return chunks;
+}
+
+function parseNumber(string) {
+	const groups = string.split(/(?=\.|e)/);
+	const wholePart = groups.filter(grp => !grp.startsWith('.') && !grp.startsWith('e'));
+	const decimalPart = groups.filter(grp => grp.startsWith('.'));
+	const powerPart = groups.filter(grp => grp.startsWith('e'));
+	return {
+		wholePart: wholePart.shift(),
+		decimalPart: decimalPart.shift().slice(1),
+		powerPart: powerPart.shift().slice(1)
+	};
+}
+
+function getFixedValue(notation, string, {minimumThreshold, preferedGroupLength}) {
+	string = removeLiteralNotation(notation, string);
+	const {wholePart, decimalPart, powerPart} = parseNumber(string);
+	const numberGroups = [];
+	const powerGroups = [];
+
+	if (wholePart) {
+		if (wholePart.length < minimumThreshold) {
+			numberGroups[0] = [wholePart];
+		} else {
+			numberGroups[0] = getChunks(wholePart, preferedGroupLength, false);
+		}
+	}
+
+	if (decimalPart) {
+		numberGroups[1] = getChunks(decimalPart, preferedGroupLength, true);
+	}
+
+	if (powerPart) {
+		if (powerPart.length < minimumThreshold) {
+			powerGroups.push(powerPart);
+		} else {
+			powerGroups.push(...getChunks(powerPart, preferedGroupLength, false));
+		}
+	}
+
+	const base = numberGroups.map(part => part.join('_')).join('.');
+
+	const power = powerGroups.length > 0 ? 'e' + powerGroups.join('_') : '';
```

Comment:
Exponential notation maybe negative.

Reply:
It already takes it into account.
If you meant the number before the `e` can be negative, then this test passes `-123_456e100`, and this `-123456e100` is correctly fixed
If you meant the power part, after the `e`, can be negative, then I haven't done any test cases but it works too. I'll add some, though.

Reply:
I mean  10000e-10000

Reply:
Then i mark this as resolve because the test `-100_000e-10_000` passes correctly 🙂

Reply:
No, you didn't ignore `-` in  this part 

`powerPart.length < minimumThreshold`

Reply:
Oh nice catch. Will fix.

Reply:
[Fixed](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/833/commits/a4c69d3c2bcd0849686e883d9e971007f5986fc0)